proxy/src/proxy.rs

```diff
@@ -124,8 +102,19 @@ pub async fn task_main(
                     }),
                 );
             }
-            _ = PROXY_CANCELLATION_TOKEN.cancelled() => {
-                bail!("Exiting");
+            maybe_exited = connections.join_next() => {
+                match maybe_exited
+                {
+                    Some(Ok(())) => {},
+                    Some(Err(e)) => { bail!(e.to_string()) },
+                    None => {
+                        if cancellation_token.is_cancelled()
+                        {
+                            drop(listener);
```

Comment:
This is unnecessary (because automatic) if you are going to exit the fn on the next line.

Reply:
Removed